test/azure/nsg_test.go

```diff
@@ -0,0 +1,91 @@
+// +build azure
+
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRangeParsingSinglePort(t *testing.T) {
```

Comment:
Please see https://github.com/gruntwork-io/terratest/pull/626/files#r490167554 about table-driven tests.

Reply:
Converted unit-tests over to table driven approach.